system/jlib/jmetrics.cpp

```diff
@@ -140,52 +153,66 @@ bool MetricsManager::addMetric(const std::shared_ptr<IMetric> &pMetric)
+            auto metaData = pMetric->queryMetaData();
+            std::unique_lock<std::mutex> lock(metricVectorMutex);
+            for (auto &metaDataIt: metaData)
+            {
+                name.append(".").append(metaDataIt.value);
+            }
 
-        auto it = metrics.find(name);
-        if (it == metrics.end())
-        {
-            metrics.insert({name, pMetric});
-            rc = true;
+            auto it = metrics.find(name);
+            if (it == metrics.end())
+            {
+                metrics.insert({name, pMetric});
+                rc = true;
+            }
+            else
+            {
+                // If there is a match only report an error if the metric has not been destroyed in the meantime
+                auto match = it->second.lock();
+                if (match)
+                {
+#ifdef _DEBUG
+                    // In debug throw an exception so the developer knows when a duplicate metric name is being added
+                    throw MakeStringException(MSGAUD_operator, "addMetric - Attempted to add duplicate named metric with name '%s'", name.c_str());
+#else
+                    // In release notify the operator of the error, but don't prevent the system from loading
+                    OERRLOG("addMetric - Adding a duplicate named metric '%s', old metric replaced", name.c_str());
+#endif
+                }
+                else
+                {
+                    rc = true;  // old metric no longer present, so it's considered unique
+                }
+                it->second = pMetric;
+            }
         }
         else
         {
-            // If there is a match only report an error if the metric has not been destroyed in the meantime
-            auto match = it->second.lock();
-            if (match)
-            {
 #ifdef _DEBUG
-                // In debug throw an exception so the developer knows when a duplicate metric name is being added
-                throw MakeStringException(MSGAUD_operator, "addMetric - Attempted to add duplicate named metric with name '%s'", name.c_str());
+            // In debug throw an exception to notify the developer of the invalid name
+            throw MakeStringException(MSGAUD_operator, "addMetric - Attempted to add metric with invalid name ('%s')", name.c_str());
 #else
-                // In relase notify the operator of the error, but don't prevent the system from loading
-                OERRLOG("addMetric - Adding a duplicate named metric '%s', old metric replaced", name.c_str());
+            // In release notify the operator of the error, but don't prevent the system from loading
+            OERRLOG("addMetric - Metric name is not valid '%s', metric not added", name.c_str());
 #endif
-            }
-            else
-            {
-                rc = true;  // old metric no longer present, so it's considered unique
-            }
-            it->second = pMetric;
         }
     }
-    else
+    catch (...)
```

Comment:
minor/question: are there any other exception classes other than std::regex that could be thrown above.

Not keen on catching ... in general, because (for example) it will also catch ISEH_Exception.
But also in this case if it's an IException (like the ones thrown in this func), this code will now lose those IException's. effectively replacing it with this new one.

Reply:
line 206 is still catching all exceptions, e.g. ISEH_Exceptions and IException's like the 2 this function could throw, and effectively replacing them with the new exception.
Should be changed to catch std::regex_error exception + ensure other original exceptions are reported or rethrown.

Reply:
Changed to catch std::regex error exception. Other exceptions thrown by the metrics code are not explicitly caught, so they are handled by any catch blocks in the caller's code (or program termination if not handled).